idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+			n=[n,n_elements(where(segarr eq max(segarr)))]
+			x=[x,xco[where(segarr eq max(segarr))]]
+			y=[y,yco[where(segarr eq max(segarr))]]
+
+			XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='000000'xL,charthick=7,charsize=2
+			XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='FFFFFF'xL,charthick=3,charsize=2
+			props[0,loc[0]+1]=strcompress(long(max(segarr)),/remove_all)
+
+		endif
+	endfor
+endif
+
+;====cycle through any CHs which was not relabeled in tracking========
+for i=1,ident do begin
+	if taken[i] ne 1 then begin
+		mxseg++
+		segarr[where(iarr eq i)]=mxseg
+
+;====contour and label CH boundary======
+		Contour,segarr,ax,ay,/over,levels=[mxseg-0.5],path_xy=xy,path_info=info,/path_data_coords;,color='FFFFFF'xL,thick=3
+		for j=0L,(n_elements(info)-1) do begin
+			chpts = [FINDGEN(info(J).N), 0]
+			PLOTS, xy(*,INFO(J).OFFSET + (10*chpts[0:(max(chpts)/10)]) ), /DATA,color='FFFFFF'xL,thick=3
+			PLOTS, xy(*,INFO(J).OFFSET), /DATA,color='FFFFFF'xL,thick=2,/continue		
+		endfor	
+		cenarr[*]=0
+		cenarr[where(segarr eq max(segarr))]=1
+		precen=centroid(cenarr)
+
+		ch=[ch,long(max(segarr))]
+		n=[n,n_elements(where(segarr eq max(segarr)))]
+		x=[x,xco[where(segarr eq max(segarr))]]
+		y=[y,yco[where(segarr eq max(segarr))]]
+
+		XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='000000'xL,charthick=7,charsize=2
+		XYOUTS, precen[0], precen[1], 'CH'+strcompress(long(max(segarr)),/remove_all),alignment=0.5,color='FFFFFF'xL,charthick=3,charsize=2
+		props[0,i+1]=strcompress(long(max(segarr)),/remove_all)
+	endif
+endfor
+
+;====display off-limb CH regions=======
+Contour,offarr,ax,ay,/over,levels=[0.5],color='FFFFFF'xL
+
+;====create image in output folder=======
+void = cgSnapshot(Position=[0.035,0.035,0.98,0.98],File=outpath+'/pngs/saia/saia_chimr_ch_'+time2file(map[1].time,/seconds)+'_pre', /PNG, /NoDialog)
+
+;====create structure containing simple CH location information======
+chim={date:ind.date_obs,index:ind,ch:ch[1:ident],n:n,x:float(x[1:n_elements(x)-1]),y:float(y[1:n_elements(y)-1]),mxseg:mxseg}
+
+;====stores all CH properties in a text file=====
+for i=2,ident+1 do begin
+	formtab[i]=props[0,i]+' '+props[1,i]+' '+props[2,i]+' '+props[3,i]+' '+props[4,i]+' '+props[5,i]+' '+props[6,i]+' '+props[7,i]+' '+props[8,i]+' '+props[9,i]+' '+props[10,i]+' '+props[11,i]+' '+props[12,i]+' '+props[13,i]+' '+props[14,i]+' '+props[15,i]+' '+props[16,i]+' '+props[17,i]+' '+props[18,i]+' '+props[19,i]+' '+props[20,i]+' '+props[21,i]+' '+props[22,i]+' '+props[23,i]
+	printf,2,formtab[i]
+endfor
+close,2
+
+;====Save CH structure to save file=====
+;save,chim,filename=outpath+'/meta/arm_ch_location_'+time2file(map[1].time)+'.sav'
+
+loadct,0,/silent
+set_plot,'x'
+;====EOF====
+if f171 eq '' or f193 eq '' or f211 eq '' or fhmi eq '' then begin
+ jump1: print, 'Not all files are present.'
+endif
```

Comment:
Yes, an early return would be preferred.
The use of `goto` is not good practice because you loose the linearity of the code and is very very difficult to debug.